forem/forem
# RFC#50-P5(a) Add feature flags on some of the inner routes

## What type of PR is this? (check all applicable)


## Description
This PR is a continuation of the admin restructure work. In this PR, I add Feature Flags to some of the inner routes (From Overview working my way downwards to Customization: HTML Variants). Whilst this is an absolutely laborious task, and looks very clunky in the codebase, I kept it manual because it will give us a true representation of what's broken/working. Honestly I'm also not sure if there was any smarter way to do this (so if you reading this and you're thinking of an idea, let me know :)!).

Note that I've still got a long way to go in order to redirect all the routes, but I didn't want to overwhelm reviewers and was hoping for a faster feedback cycle so I'm batching them into possibly 3 PR's.

What do I not do in this PR: 
- There have been some renders like `render :new` which is not playing nicely with a scoped route and so I'm still figuring that out. I've got a list of these and they appear in the  HTML Variants controller and tags controller.
- The sidebar does not stay open and highlighted for an inner route. I'm working on this. 

If you're curious about whats next and concerned about when these ugly feature flags will be removed, heres the next steps:
- In the next day or two I will push the completed version out (still enabled under the feature flag).
- Before the end of the week, we will flip the feature flag to be true (so that we view the new admin navbar) on a smaller but used Forem (maybe https://forem.team/) where most people have access.
- I'll then ask the community success team + the rest of the team to test and use the new admin interface to make changes.
- The team can test next week whilst I'll be away.
- Once they done testing, I'll address any feedback, remove the feature flags + the old code (all these else statements). This should be around the first week of April. 

## Related Tickets & Documents
https://github.com/forem/rfcs/pull/50
and 4 other earlier PR"s

## QA Instructions, Screenshots, Recordings
- Double check that I've mapped out the equivalent scoped route without any mistakes and that the else statement matches the original routte.
- Click around the interface by [enabling the admin_restructure feature flag](https://forem-team.slack.com/archives/C7GE84DEJ/p1615240364120400). Note the description section above for details on what I've covered in this PR. 

### UI accessibility concerns?
None

## Added tests?

- [x] No, I need to still update the tests o reflect new paths but that will have to be a separate endeavour. For now I'm making sure that none of the current tests fail. 

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
- [x] This change does not need to be communicated, and this is why not: It will be communicated internally once ready to test.

## [optional] Are there any post deployment tasks we need to perform?
None

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/sRH72BadEmjf3QKAAQ/giphy.gif)
